src/solvers/flattening/arrays.cpp

```diff
@@ -580,7 +580,7 @@ void arrayst::add_array_constraints_update(
 
     if(index_expr.type()!=value.type())
     {
-      prop.error() << expr.pretty() << messaget::eom;
+      prop.message.error() << expr.pretty() << messaget::eom;
```

Comment:
This change (inside a `#if 0` block) looks suspicious: I believe this should be just `error()`.